mergefs.go

```diff
@@ -25,3 +28,64 @@ func (mfs MergedFS) Open(name string) (fs.File, error) {
 	}
 	return nil, os.ErrNotExist
 }
+
+// ReadDir reads from the directory, and produces a DirEntry array of different
+// directories.
+//
+// It iterates through all different filesystems that exist in the mfs MergeFS
+// filesystem slice and it identifies overlapping directories that exist in different
+// filesystems
+func (mfs MergedFS) ReadDir(name string) ([]fs.DirEntry, error) {
+	dirsMap := make(map[string]fs.DirEntry)
+	for _, filesystem := range mfs.filesystems {
+		if fsys, ok := filesystem.(fs.ReadDirFS); ok {
+			dir, err := fsys.ReadDir(name)
+			if err != nil {
+				if errors.Is(err, fs.ErrNotExist) {
```

Comment:
If all filesystems return fs.ErrNotExist, shouldn't this function also return fs.ErrNotExist?

I'd just count the fs.ErrNotExists . `if len(mfs.filesystems) == notExistCount { return fs.ErrNotExist }`

Reply:
Hm, not entirely sure - this will just return an empty `[]fs.DirEntry` and will be skipped by next processes. Will make the change though.

Reply:
thanks. I'm talking about the directory itself not existing rather than an empty directory. All good